zenoh-ext/src/serialization.rs

```diff
@@ -33,18 +35,58 @@ impl fmt::Display for ZDeserializeError {
 }
 impl std::error::Error for ZDeserializeError {}
 
-fn default_serialize_n<T: Serialize>(slice: &[T], serializer: &mut ZSerializer) {
+#[derive(Debug)]
+pub struct ZSerializeError(String);
```

Comment:
Why not using a ZST as it is the case for deserialization?

Reply:
Why not a `Box<dyn Error + Send + Sync>`?

Reply:
What is a ZST ? About Box - yes, agree, fixing

Reply:
made it ZST for now